### ISSUE TITLE
oc apply with label selector returns error if oc process output is piped to it

### ISSUE THREAD

**Issue by tinexw** [openshift/origin#19746]
I want to create objects from a template by processing the template, piping the output to oc apply and pruning everything matching a certain label:

``oc process -f <filename> | oc apply --prune -l foo=bar -f -``

This fails with the error ``error: object does not implement the Object interfaces`` (Same as  #14450 but I couldn't reopen that issue.)

Note that it's not related to the prune option, it also fails with the same error without it, s. steps below.

##### Version
```
oc v3.7.2+282e43f
kubernetes v1.7.6+a08f5eeb62
features: Basic-Auth

Server https://192.168.64.2:8443
openshift v3.7.1+282e43f-42
kubernetes v1.7.6+a08f5eeb62
```

##### Steps To Reproduce
1. Create a template file ``test-template.yml``

```
apiVersion: v1
kind: Template
metadata:
  name: test
objects:
- apiVersion: v1
  kind: ImageStream
  metadata:
    labels:
      foo: bar
    name: test
```
2. Process the template and pipe the output to oc apply with a label selector:
``oc process -f test-template.yml | oc apply -l foo=bar -f -``

##### Current Result
``error: object does not implement the Object interfaces``

##### Expected Result
That the resources matching the label are configured: ``imagestream "test" configured``
Basically the same as if I would store all object configurations in a directory and run the apply command.

**Comment by mfojtik** [comment]
Fix: https://github.com/kubernetes/kubernetes/pull/63966

Will pick it to master once it merges in upstream.
